File: LeetCode/Interview Questions/JAVA/TwoSum.java
class Solution {
    //One pass with HashMap
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i< nums.length;i++){
            int complemento = target - nums[i];  
            if(map.containsKey(complemento)) {
                return new int[] {map.get(complemento), i};
             }
            map.put(nums[i], i); 
        }return null;
    }
}

File: LeetCode/Interview Questions/JS/TwoSum.js
let twoSum  = (nums, target) => {
    const comp = {};
    for(let i=0; i< nums.length; i++){
        if(comp[nums[i] ]>=0){
            return [ comp[nums[i] ], i];
        }
        comp[target-nums[i]] = i;
    }
}; 

let array = [2,7,11,15];
console.log(twoSum(array, 9));